caffe2/sgd/adagrad_op_gpu.cu

```diff
@@ -36,6 +36,40 @@ void adagrad_update<CUDAContext>(
       context->cuda_stream()>>>(N, g, h, ng, nh, epsilon, lr);
 }
 
+__global__ void AdagradCompute(
+    int N,
+    const float* w,
+    const float* g,
+    const float* h,
+    float* nw,
+    float* nh,
+    float epsilon,
```

Comment:
normally outputs passed via  a pointer go in the end (according to google code style which we generally use)